frontend/src/views/QuestionPageView.tsx

```diff
@@ -21,6 +25,34 @@ const QuestionPageView: React.FC = () => {
     setShowDropdown(false);
   };
 
+  useEffect(() => {
+    const fetchUsername = async (uid: string) => {
+      try {
+        
+        const usernameData = await getUsernameByUid(uid);
+        setUsername(usernameData);
+      } catch (error) {
+        console.error("Failed to retrieve username:", error);
+        setUsername("");
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    const token = sessionStorage.getItem("authToken");
+    if (token) {
+      const decoded = decodeToken(token);
+      if (decoded && decoded.user_id) {
+        
+        fetchUsername(decoded.user_id);
```

Comment:
Do not need to decode token to get the uid. Obtain uid via `const uid = sessionStorage.getItem("uid");`

Perhaps this might also mean you can delete the `decodeToken` function.

Reply:
Good catch, will update!